dotnet/src/Connectors/Connectors.Memory.Common/SqlFilterTranslator.cs

```diff
@@ -178,30 +181,33 @@ private void TranslateMember(MemberExpression memberExpression, bool isSearchCon
         switch (memberExpression)
         {
             case var _ when this.TryBindProperty(memberExpression, out var property):
-                this.GenerateColumn(property.StorageName, isSearchCondition);
+                this.GenerateColumn(property, isSearchCondition);
                 return;
 
             case var _ when TryGetCapturedValue(memberExpression, out var name, out var value):
-                this.TranslateCapturedVariable(name, value);
+                this.TranslateCapturedVariable(value);
                 return;
 
             default:
                 throw new NotSupportedException($"Member access for '{memberExpression.Member.Name}' is unsupported - only member access over the filter parameter are supported");
         }
     }
 
-    protected virtual void GenerateColumn(string column, bool isSearchCondition = false)
-        => this._sql.Append('"').Append(column.Replace("\"", "\"\"")).Append('"');
+    protected virtual void GenerateColumn(VectorStoreRecordPropertyModel column, bool isSearchCondition = false)
+        // all SQL-based connectors are required to escape values returned by StorageName property
+        => this._sql.Append('"').Append(column.StorageName).Append('"');
 
-    protected abstract void TranslateCapturedVariable(string name, object? capturedValue);
+    // Don't use the variable name to avoid issues with escaping and quoting.
+    // The derived types are supposed to create the parameter name from a prefix and _parameterIndex.
```

Comment:
I decided that instead of passing the captured variable name and creating a connector-specific parameter name from it, it will be easier to just always create the names based on parameter index. It's not as pretty as having the names, but it's very simple and foremost safe.

```diff
- paramName = $"@{variableName}{ _parameterIndex++}"
+ paramName = $"@{_parameterIndex++}"
```

Reply:
This is indeed simpler, but the generated SQL would be significantly less readable... The SQL shows up in logs/telemetry/db perf tools, and we'd ideally do our best to keep it as readable as possible.

BTW it shouldn't be extremely hard to do this in a way that's widely compatible - the logic we use for this in EF is simply taking the variable name, and then adding a counter only if needed. Here's what I'm proposing in #11750 ([link](https://github.com/microsoft/semantic-kernel/pull/11750/files#diff-3143f1a10c972014aff482b1d20551b92ca18b12a3a896f269984048eff97385R115)):

```c#
var origName = memberInfo.Name;
var name = origName;
for (var i = 0; this._parameterNames!.Contains(name); i++)
{
    name = $"{origName}_{i}";
}
this._parameterNames.Add(name);
```

Reply:
> simply taking the variable name, and then adding a counter only if needed

But a C# (not to mention F# or IL that can contain almost any character) variable name can contain a character that is not valid in SQL parameter name. Example: any character like `ą` (polish) or `ä` (german) is not valid for Sqlite.

I know it's unlikely and I may be a little paranoid (side effect of working on BinaryFormatter), but I wanted to it to be always safe and correct.

Reply:
However, I could so the same thing I did for SQL Server: stop on first non-ascii character https://github.com/microsoft/semantic-kernel/blob/2ec7d24222afa74d9673e7773902b27f103b857a/dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerCommandBuilder.cs#L439-L445

Reply:
These are good points indeed... I can only say that in all the time handling EF issues, I don't recall ever seeing a problem here - probably nobody has tried to use weird characters in C# variable names etc.

But your proposal to stop on the first non-ascii character sounds like a good compromise - in all normal cases it will do nothing at all (since all characters will be ascii), but in case somebody does do it, it will kick in.

So up to you... I think it's OK for us to GA without this very high level of safety - but OK to add it!

Reply:
I've tried to apply this suggestion, but it turned out that for both Postgres and SqlServer some other parts of the code rely on this simple naming `@number`:

https://github.com/microsoft/semantic-kernel/blob/c8b129df5d185116f836effd73d648d82f59ab62/dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresFilterTranslator.cs#L52-L53

https://github.com/microsoft/semantic-kernel/blob/c8b129df5d185116f836effd73d648d82f59ab62/dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerFilterTranslator.cs#L102-L104

I decided to keep it as is.

Reply:
You don't think we should try to preserve the name where we can? Debugging/tracking down SQL performance issues can be quite hard if you can't read the SQL easily...

Reply:
I am not saying we should not at all. It's just right now I don't have the time to unify all the 3 connectors in that regard.

Reply:
OK. If we need to deliver an imperfect thing (which is totally fine), I'd personally go the other way, i.e. produce more readable SQL for the 99% case while failing for the tiny minority case which have captured variable names with weird characters... But I'm OK to go like this as well.